### PR TITLE
Add framework for testing nalu-wind-nightly package

### PR DESCRIPTION
**Pull request by tasmith4**
* For now, we don't concretize the spec, in order to make the test predictable and robust -- i.e. we don't want to be checking for specific variants, and then fail later when someone makes another variant default.
  * Not concretizing has the added advantage of not slowing the tests down by several seconds per spec.
  * The fixture does memoize packages, so we only have to create one for each unique abstract spec if we do decide we really need concretization.
* Also, this adds *.swp to the .gitignore

First step towards resolving #280

### CODE REVIEW

File: spack-scripting/tests/test_nalu_wind_nightly.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import pytest
+import importlib
+
+nalu_wind_nightly = importlib.import_module('...repos.exawind.packages.nalu-wind-nightly.package', 'package')
+#from ...repos.exawind.packages.nalu-wind-nightly.package import *
```
Comment (psakievich) [inline spack-scripting/tests/test_nalu_wind_nightly.py]:
```suggestion
from spack.pkg.exawind.nalu_wind_nightly import NaluWindNighlty 
```
Have you tried this?

File: spack-scripting/tests/conftest.py [hunk 2]
```diff
@@ -1 +1,34 @@
...
+import pytest
+
+import spack.repo
+from spack.spec import Spec
 from spack.test.conftest import *  # noqa: F401
+
+
+@pytest.fixture(scope='session')
+def builtin_repo_path():
+    yield os.path.join(os.environ['SPACK_MANAGER'], 'spack', 'var', 'spack', 'repos', 'builtin')
```
Comment (psakievich) [inline spack-scripting/tests/conftest.py]: This is already defined with `spack.paths.packages_path`

Comment (tasmith4) [reply spack-scripting/tests/conftest.py]: fixed, thanks

**Comment by tasmith4** [comment]
@psakievich I thought it would be good to get this testing framework into the repo, independent of the other bug fix, so I'm marking as ready for review now.

**Review by psakievich — APPROVED** [review]
Looks great! Thanks for doing this

**Pull request merged**